spring-integration-core/src/main/java/org/springframework/integration/aop/CompoundTriggerAdvice.java

```diff
@@ -47,8 +51,21 @@ public CompoundTriggerAdvice(CompoundTrigger compoundTrigger, Trigger overrideTr
 		this.override = overrideTrigger;
 	}
 
+	/**
+	 * @param result the received message.
+	 * @param source the message source.
+	 * @return the message or null
+	 * @deprecated since 5.3 in favor of {@link #afterReceive(Message, Object)}
+	 */
 	@Override
+	@Deprecated
 	public Message<?> afterReceive(Message<?> result, MessageSource<?> source) {
+		return afterReceive(result, (Object) source);
+	}
+
+	@Override
+	@Nullable
+	public Message<?> afterReceive(@Nullable Message<?> result, Object source) {
```

Comment:
Wouldn't something like `Advisable<T>` be better than `Object`?

Reply:
Well, in most cases it is really about both `MessageSource` and `PollableChannel`, so even with generic argument it still going to be an `Object` 😄 

I thought about a `Supplier` for both those interfaces, but this is not what we are going to use with the `receive(timeout)`.
Therefore I'm OK to stick with an `Object` for a `source` argument.